src/common/ErrorAccumulator.h

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2019 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/dkfz-odcf/FastqIndEx/blob/master/LICENSE.txt).
+ */
+
+#ifndef FASTQINDEX_ERRORACCUMULATOR_H
+#define FASTQINDEX_ERRORACCUMULATOR_H
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+using std::vector;
+using std::string;
+
+
+/**
+ * The class is used to provide basic error storage methods.
+ */
+class ErrorAccumulator {
```

Comment:
I don't get my head around this class. It is half static (use of cerr) and half not (errorMessages). Almost like a singleton in some sense, maybe more like a static Service class. 

I think we should discuss whether could make the usage of this class more obvious. Maybe split? Maybe make it a real Service class with a single instance in a field of a `Services` singleton, something like that.